src/Features/Core/Portable/CodeCleanup/readme.md

```diff
@@ -0,0 +1,31 @@
+- Types Implemented by Visual Studio used by Roslyn
+  - Attributes used to add data to a [MEF](https://docs.microsoft.com/dotnet/framework/mef/) export of a `FixId`
+    - [FixIdAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.fixidattribute) as string representing the identify of a something that an [ICodeCleanUpFixer](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupfixer) can fix. We use the Diagnostic Id (such as `IDE001`) for these values.
+    - [NameAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.utilities.nameattribute) as string representing the name of something that an [ICodeCleanUpFixer](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupfixer) can fix. We also use the Diagnostic Id (such as `IDE001`) for these values since we don't have a concept of an Id being different than the Name.
+    - [OrderAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.utilities.orderattribute) an attribute used to order items in the UI. You can ask for an item to be ordered before or after something by referring to its `FixId` (the string given to the [FixIdAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.fixidattribute))
+    - [ConfigurationKeyAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.configurationkeyattribute) unused by us, we have our own configuration mechanisms for roslyn. We always export a [ConfigurationKeyAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.configurationkeyattribute) with the value of `"unused"`.
+    - [HelpLinkAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.helplinkattribute) url to direct the user to. should be a page on docs.microsoft.com.
+    - [LocalizedNameAttribute](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.utilities.localizednameattribute) The string that will be used in the VS UI.
+  - [ICodeCleanUpScope](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupscope) exchange type passed to our [ICodeCleanUpFixer](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupfixer) implementation used to tell us what to fix (document, project, solution, etc)
+  - [ICodeCleanUpExecutionContext](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupexecutioncontext)exchange type passed to our [ICodeCleanUpFixer](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.language.codecleanup.icodecleanupfixer) implementation used to tell us what `FixId`s (aka Diagnostic Ids for us) that the user has chosen to fix in the UI.
+
+## How these types fit together
```

Comment:
Would love to see a kind of step by step "I added a new thing, how do I make it appear in code cleanup" section in this readme